pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -232,15 +241,62 @@ func (c *Reconciler) finishReconcileUpdateEmitEvents(ctx context.Context, pr *v1
 	return merr
 }
 
-func (c *Reconciler) updatePipelineResults(ctx context.Context, pr *v1beta1.PipelineRun) {
-	logger := logging.FromContext(ctx)
+// resolvePipelineState will attempt to resolve each referenced task in the pipeline's spec and all of the resources
+// specified by those tasks.
+func (c *Reconciler) resolvePipelineState(
+	ctx context.Context,
+	tasks []v1beta1.PipelineTask,
+	pipelineMeta *metav1.ObjectMeta,
+	pr *v1beta1.PipelineRun,
+	providedResources map[string]*v1alpha1.PipelineResource) (resources.PipelineRunState, error) {
+	pst := resources.PipelineRunState{}
+	// Resolve each task individually because they each could have a different reference context (remote or local).
+	for _, task := range tasks {
+		fn, _, err := tresources.GetTaskFunc(c.KubeClientSet, c.PipelineClientSet, task.TaskRef, pr.Namespace, pr.Spec.ServiceAccountName)
```

Comment:
Same comment as `GetPipelineFunc`, this can be part of `ResolvePipelineRunTask` where a task is retrieved/resolved based on `taskRef` or `taskSpec`.

Reply:
For the same reason as above, I am inclined to push this to a larger refactor. https://github.com/tektoncd/pipeline/issues/3305 to capture that work.